### PR TITLE
Payments

### PR DESCRIPTION
**Pull request by Chimexx**
-endpoint to get item price from db.
-endpoint to verify paystack transaction 

-Once a user clicks on payment button, a request is made to the getPrice endpoint, which returns price of the item.
-On success, paystack proceeds with the transaction and returns a reference.
-The reference is sent to the payment verifcation endpoint, which will check for connectivity issues as well as the transaction status, and returns the status.
-Frontend will now proceed depending on the received status.

### CODE REVIEW

**Review by andyjrII — APPROVED** [review]

File: src/payments/payments.service.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      )
+        .then((success) => {
+          result = success;
+        })
+        .catch((error) => {
+          result = error;
+        });
+
+      /* This is a check for internet connectivity issues. */
+      if (!result.response && result.status !== 200)
```
Comment (SunmadeBabafemi) [inline src/payments/payments.service.ts]: if result is made the response object of the axios request.. i think result.response is not a part of the result contents ... 
here are the list of parameters in an axios response object
data - the payload returned from the server
status - the HTTP code returned from the server
statusText - the HTTP status message returned by the server
headers - headers sent by server
config - the original request configuration
request - the request object

... perhaps maybe what you should be looking for is -> result.data

**Review by PhilemonBrain — APPROVED** [review]

File: src/payments/payments.service.ts (hunk 1 above)
Comment (PhilemonBrain) [reply src/payments/payments.service.ts]: @Chimexx can u confirm this b4 i merge this

Comment (Chimexx) [reply src/payments/payments.service.ts]: @PhilemonBrain, He's right and it's fixed

Comment (PhilemonBrain) [reply src/payments/payments.service.ts]: @Chimexx what do you think about babafemis comment on result.reponse

Comment (Chimexx) [reply src/payments/payments.service.ts]: I said he's right, and I've pushed an update

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**